### PR TITLE
Fix issues reported by cppcheck.

### PR DESCRIPTION
**Pull request by antiagainst**
Remove code not being used, add explicit to constructors, and
add missing fields in constructors.

### CODE REVIEW

File: source/binary.cpp [hunk 1]
```diff
@@ -62,47 +62,6 @@ spv_result_t spvBinaryHeaderGet(const spv_const_binary binary,
   return SPV_SUCCESS;
 }
 
-// TODO(dneto): This API is not powerful enough in the case that the
-// number and type of operands are not known until partway through parsing
-// the operation.  This happens when enum operands might have different number
-// of operands, or with extended instructions.
-spv_operand_type_t spvBinaryOperandInfo(const uint32_t word,
```
Comment (dneto0) [inline source/binary.cpp]: I'm glad this API is going.  But we should also remove the code in validate.cpp between an #if 0 / #endif pair that references it, as well the prototype in binary.h

File: source/opcode.cpp [hunk 2]
```diff
@@ -485,16 +478,6 @@ int32_t spvOpcodeIsPointer(const SpvOp opcode) {
   }
 }
 
-int32_t spvInstructionIsInBasicBlock(const spv_instruction_t* pFirstInst,
```
Comment (dneto0) [inline source/opcode.cpp]: Should also remove the prototype for this in opcode.h

File: source/text_handler.h [hunk 3]
```diff
@@ -263,10 +263,9 @@ class AssemblyContext {
     // It also allows octal input, but we don't care about that case.
     text_stream >> std::setbase(0);
     text_stream >> *value_pointer;
-    bool ok = true;
 
     // We should have read something.
-    ok = (text[0] != 0) && !text_stream.bad();
+    bool ok = (text[0] != 0) && !text_stream.bad();
```
Comment (dneto0) [inline source/text_handler.h]: Something complained about this?  Weird.  I think any decent front end would collapse this.  But ok.

**Comment by dneto0** [comment]
Please delete even more code.
-1

File: source/binary.cpp (hunk 1 above)
Comment (antiagainst) [inline source/binary.cpp]: Done.

File: source/opcode.cpp (hunk 2 above)
Comment (antiagainst) [inline source/opcode.cpp]: Done.

File: source/text_handler.h (hunk 3 above)
Comment (antiagainst) [inline source/text_handler.h]: Right. Kinda wired for me. But whatever.

Comment (dekimir) [inline source/text_handler.h]: Not weird at all -- in fact, the style guide [requires](https://google.github.io/styleguide/cppguide.html#Local_Variables) it.

**Comment by dekimir** [comment]
+1

**Comment by dneto0** [comment]
+2

**Pull request merged**